cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Cli/Conway/DRepRetirement.hs

```diff
@@ -0,0 +1,298 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeApplications #-}
+
+module Cardano.Testnet.Test.Cli.Conway.DRepRetirement
+  ( hprop_drep_retirement
+  ) where
+
+import           Control.Monad (void)
+import           Control.Monad.IO.Class
+import           Control.Monad.Trans.Except (runExceptT)
+import           Control.Monad.Trans.State.Strict (put)
+import qualified Data.Aeson as Aeson
+import           Data.Data
+import qualified Data.Map.Strict as Map
+import qualified Data.Text as Text
+import           Data.Type.Equality (testEquality)
+import qualified Data.Vector as V
+import           GHC.IORef (newIORef)
+import           GHC.Stack (HasCallStack, withFrozenCallStack)
+import qualified Hedgehog as H
+import           Hedgehog.Extras (Integration)
+import           Lens.Micro ((^.))
+import           Prelude
+import           System.FilePath ((</>))
+
+import qualified Cardano.Api as Api
+
+import           Cardano.Api
+import           Cardano.Api.Ledger
+import qualified Cardano.Api.Ledger as L
+import           Cardano.Api.Pretty (docToString)
+import           Cardano.CLI.Types.Output (QueryTipLocalStateOutput (..))
+import qualified Cardano.Ledger.Shelley.LedgerState as L
+import           Cardano.Testnet
+import           Control.Monad.Catch (MonadCatch)
+import           Hedgehog
+import qualified Hedgehog.Extras as H
+import qualified Hedgehog.Extras.Stock.IO.Network.Sprocket as IO
+import qualified Testnet.Process.Cli as P
+import           Testnet.Process.Cli (execCliStdoutToJson)
+import qualified Testnet.Process.Run as H
+import qualified Testnet.Property.Utils as H
+import           Testnet.Property.Utils (queryUtxos)
+import           Testnet.Runtime
+
+-- | The goal of this test is to check the property 'DRep Registration/DeRegistration', coming from this document:
+-- https://docs.google.com/document/d/1qS5Z7vTNZTlvP66sLuYyhGkjU4rAsO92urDHnEkvJMQ/edit?pli=1#heading=h.9pj6u2uc0i4q
+--
+-- Execute this test with:
+-- @DISABLE_RETRIES=1 cabal test cardano-testnet-test --test-options '-p "/DRepRetirement/"'@
+hprop_drep_retirement :: Property
+hprop_drep_retirement = H.integrationRetryWorkspace 2 "drep-retirement" $ \tempAbsBasePath' -> do
+  -- Start a local test net
+  conf@Conf { tempAbsPath } <- H.noteShowM $ mkConf tempAbsBasePath'
+  let tempAbsPath' = unTmpAbsPath tempAbsPath
+      tempBaseAbsPath = makeTmpBaseAbsPath tempAbsPath
+
+  work <- H.createDirectoryIfMissing $ tempAbsPath' </> "work"
+
+  utxoFileCounter <- liftIO $ newIORef 1
+
+  let sbe = ShelleyBasedEraConway
+      era = toCardanoEra sbe
+      cardanoNodeEra = AnyCardanoEra era
+      fastTestnetOptions = cardanoDefaultTestnetOptions
+        { cardanoEpochLength = 100 -- 100ms epoch
```

Comment:
100 slot epoch which would be a 10 second epoch.

Reply:
Updated the comment :+1: